### PR TITLE
chore(pagerduty) Align pagerduty config implementation with opsgenie

### PR DESCRIPTION
**Pull request by markstory**
The opsgenie integration uses module functions to manage its keyring data. Align pagerduty with this implementation so that we don't have a handful of pagerduty methods attached to OrganizationIntegration.

Longer term I'd like to normalize the logic that opsgenie and pagerduty as it is largely the same outside of naming conventions.

### CODE REVIEW

File: src/sentry/integrations/pagerduty/utils.py [hunk 1]
```diff
@@ -20,6 +19,64 @@
...
+class PagerDutyServiceDict(TypedDict):
+    integration_id: int
+    integration_key: str
+    service_name: str
+    id: int
+
+
+def add_service(
+    organization_integration: OrganizationIntegration, integration_key: str, service_name: str
+) -> PagerDutyServiceDict:
```
Comment (leeandher) [inline src/sentry/integrations/pagerduty/utils.py:31]: We should tag this with the `@control_silo_function` decorator

File: src/sentry/services/hybrid_cloud/integration/service.py [hunk 2]
```diff
@@ -132,19 +128,6 @@ def get_organization_integration(
...
-        org_integration = self.get_organization_integration(
-            integration_id=integration_id, organization_id=organization_id
-        )
-        if not org_integration:
-            return None
-        try:
-            return OrganizationIntegration.find_service(org_integration.config, service_id)
-        except StopIteration:
-            return None
-
```
Comment (leeandher) [inline src/sentry/services/hybrid_cloud/integration/service.py:147]: We're not rolled out so it doesn't matter, but in the future would we be able to whole cloth remove RPC services like this? This gets me thinking about deprecation strategies for stuff like this down the line 🤔

**Review by leeandher — APPROVED** [review]
Looks great! All for better standards across integrations 👏

File: src/sentry/services/hybrid_cloud/integration/service.py (hunk 2 above)
Comment (markstory) [reply src/sentry/services/hybrid_cloud/integration/service.py:147]: > We're not rolled out so it doesn't matter, but in the future would we be able to whole cloth remove RPC services like this? 

No we won't be able to remove RPC methods soon. Thankfully this wasn't an RPC method.

In the future we'd need to stop calling an RPC method, deploy that change, then deploy a separate change to remove the RPC method. Much like migrations removals will always need to be done over multiple deploys.

**Review by dashed — APPROVED** [review]

**Pull request merged**